diff --git a/device/sprd/sharkle/common/security_feature.mk b/device/sprd/sharkle/common/security_feature.mk
index aff7e6b..1a1782e 100755
--- a/device/sprd/sharkle/common/security_feature.mk
+++ b/device/sprd/sharkle/common/security_feature.mk
@@ -249,3 +249,13 @@ BOARD_SEPOLICY_DIRS += vendor/sprd/proprietories-source/sprdtrusty/vendor/sprd/m
 endif
 
 endif
+
+ifeq ($(BOARD_TRUSTKERNEL_TEE), true)
+TRUSTKERNEL_TEE_SUPPORT := yes
+
+$(call inherit-product, \
+   vendor/sprd/proprietories-source/trustkernel/source/bsp/platform/common/scripts/device_trustkernel.mk)
+
+BOARD_SEPOLICY_DIRS += \
+   vendor/sprd/proprietories-source/trustkernel/source/bsp/platform/common/scripts/sepolicy/Android9.0
+endif
